README.md

```diff
@@ -16,8 +16,12 @@
 Add your answers to this markdown file.
 
 
-1. One reason asymptotic analysis may be misleading is when using smaller inputs such as a small array that needs to be sorted. When doing asysmptotic analysis, we ignore constant factors, but these constant factors can make a big difference to the run time if we are dealing with a smaller array. Another reason asymptotic analysis may be misleading is that it can assume that we do not have any knowledge of the data. For example, we may collect data that is already sorted or close to sorted. So if we want to choose a sorting algorithm to sort that data, we may look at the average case run times for all sorting algorithms and choose the fastest one. But if we know that the data is already sorted or close to sorted, we could choose an algorithm like insertion sort which has one of the worst average case run times but one of the best best case run times. Finally, a third reason why asymptotic analysis can be misleading is that it ignores hardware factors. The run time for certain algorithms may be faster, but require more memory from the device. This could cause problems if you want to work with a faster algorithm on a device that does not have enough memory to support that algorithm.
+1. One reason asymptotic analysis may be misleading is that we ignore constant factors within the algorithms, but constant factors can have a large effect on the overall runtime. For example, a function could have a linear time complexity, but a function that has runtime 1(n) vs 1000(n) could be a significant difference in runtime, yet bot functions have the same time complexity. Another reason asymptotic analysis may be misleading is because the bounds are very loose, so the overall time complexity is not very accurate in runtime prediction. Finally, a third reason why asymptotic analysis can be misleading is that it ignores hardware factors. A computer that has a computer with a lot of computational power will perform significantly faster with the same algorithm as a computer that has very poor computational power. Hardware factors could be constant or exponential factors that significantly slow down the runtime of a function. For example, a computer that has a very complex and efficient arithmetic logic unit can perform mathematic operations very fast. This versus a computer that has a very rudimentary and much less efficient arithmetic logic unit could have an exponential effect on the runtime of an algorithm.
```

Comment:
Not following your example -- being able to do arithmetic operations would only ever account for a constant factor, wouldn't it?

Reply:
If we took very simple arithmetic operations, and both the ALU's performed them in about the same time, because the problem is so simple to do, both ALU's can perform them in an efficient time. However, if we take larger inputs and more complex arithmetic operations, and the more complex ALU far out-performs the rudimentary ALU, wouldn't that result in an exponential difference of runtimes?

Reply:
I don't see how it would. Can you give a reference for that?

Reply:
I was just reading about ALU's and FPU's and I came up with that theory. I did not have a specific source for the theory.